### PR TITLE
Add maven.exec.xxx.skip option for exec-maven-plugin

### PR DESCRIPTION
**Pull request by asdf2014**
Try to fix [#6160](https://github.com/apache/incubator-druid/issues/6160)

### CODE REVIEW

**Comment by asdf2014** [comment]
It seems that those maven options are in effect. The travis job#[415338345](https://travis-ci.org/apache/incubator-druid/jobs/415338345) failed again related to the test case timeout. Would you please help me to rebuild it? @jihoonson

**Comment by jihoonson** [comment]
@asdf2014 thanks for taking this issue. I've restarted the failed one. 

BTW, IMO, it would be better to have an option to make the distribution tarball rather than avoiding pulling some dependencies. As you can see in https://github.com/apache/incubator-druid/blob/master/distribution/pom.xml, `mvn install` does more things other than pulling libraries to build the distribution tarball. I think we can simply skip this if some option like `-Pdist` is not specified. Please check [Hadoop's pom.xml](https://github.com/apache/hadoop/blob/trunk/hadoop-dist/pom.xml#L148-L226) as an example.

**Comment by asdf2014** [comment]
Hi, @jihoonson . Thanks for your help. The travis job is succeeded. Yes, using profile to control this will be better, it has been changed.

**Comment by jihoonson** [comment]
Thanks. Would you please add a doc for this? Probably you can add it to https://github.com/apache/incubator-druid/blob/master/docs/content/development/build.md.

Also, if you want, you can change the travis script to not make the distribution tarball, so that we can check it helps Travis jobs to succeed.

**Comment by asdf2014** [comment]
@jihoonson Indeed, it is necessary to double check it. After I added this option, it seems that the `-Pdist` option works, and the `exec-maven-plugin` plugin starts [running](https://travis-ci.org/apache/incubator-druid/jobs/416639456#L862).

File: distribution/pom.xml [hunk 1]
```diff
@@ -174,6 +83,111 @@
         </plugins>
     </build>
     <profiles>
+        <profile>
+            <id>dist</id>
+            <activation>
+                <activeByDefault>false</activeByDefault>
+                <property>
+                    <name>tar</name>
```
Comment (jihoonson) [inline distribution/pom.xml]: Would you explain how this property is being used?

File: docs/content/development/build.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ mvn clean package
 This will compile the project and create the Druid binary distribution tar under
 `distribution/target/druid-VERSION-bin.tar.gz`.
 
-This will also create a tarball that contains `mysql-metadata-storage` extension under 
+After adding `-Pdist` option, this will also create a tarball that contains `mysql-metadata-storage` extension under
```
Comment (jihoonson) [inline docs/content/development/build.md]: We need to add more details about this option. The document should say what this option means and how different `mvn clean install` and `mvn clean install -Pdist` are. Also, please update [this command](https://github.com/apache/incubator-druid/pull/6162/files#diff-15788e24b4299e22aff8aba2ea46281fR19) to `mvn clean install -Pdist`

File: distribution/pom.xml (hunk 1 above)
Comment (asdf2014) [reply distribution/pom.xml]: Sure, we also can use `-Dtar` to execute this plugin. In addition, if we used both of `name` and `value` tags, then only `-Dtar=dist` will be in effect.

```xml
<property>  
    <name>tar</name>  
    <value>dist</value>  
</property>
```

File: distribution/pom.xml [hunk 3]
```diff
@@ -174,6 +83,111 @@
         </plugins>
     </build>
     <profiles>
+        <profile>
```
[the remark this answers is not in the document]
Comment (asdf2014) [reply distribution/pom.xml]: Do you mean that the `bundle-contrib-exts` part should also be included in the `dist` profile? Or should we include all the  `build` plugins, such as the `maven-assembly-plugin` and `license-maven-plugin` plugins?

File: docs/content/development/build.md (hunk 2 above)
Comment (asdf2014) [reply docs/content/development/build.md]: Okay, i will improve this.

File: distribution/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jihoonson) [reply distribution/pom.xml]: I mean both. We don't have to execute assembling the tarball or downloading licenses for the `install` phase. Also, `bundle-contrib-exts` is currently not executed by default, but it should also be a part of building the distribution tarball.

[the remark this answers is not in the document]
Comment (asdf2014) [reply distribution/pom.xml]: Okay, i got it.

File: distribution/pom.xml (hunk 1 above)
Comment (asdf2014) [reply distribution/pom.xml]: Okay.

Comment (asdf2014) [reply distribution/pom.xml]: Added.

File: docs/content/development/build.md (hunk 2 above)
Comment (asdf2014) [reply docs/content/development/build.md]: Done.

File: distribution/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (asdf2014) [reply distribution/pom.xml]: Done.

**Comment by jihoonson** [comment]
Thanks @asdf2014. Would you check this warning which is seen when building the project.

```

File: docs/content/development/build.md [hunk 4]
```diff
@@ -13,16 +13,18 @@ Building Druid requires the following:
...
 cd druid
-mvn clean package
+mvn clean package -Pdist
 ```
 
 This will compile the project and create the Druid binary distribution tar under
 `distribution/target/druid-VERSION-bin.tar.gz`.
 
-This will also create a tarball that contains `mysql-metadata-storage` extension under 
+The difference between `mvn clean package` and `mvn clean package -Pdist` is that
```
Comment (jihoonson) [inline docs/content/development/build.md]: How about this? https://gist.github.com/jihoonson/bffa89fd2a5f9473bb6467d3c52f12f6

Comment (asdf2014) [reply docs/content/development/build.md]: Hi, @jihoonson . Sorry for the delayed reply, this has been patched, PTAL.

**Comment by jihoonson** [comment]
@asdf2014 sorry for the delayed review. Would you please resolve conflicts? I'll review again.

**Comment by asdf2014** [comment]
Hi, @jihoonson . The conflict has been fixed. PTAL.

File: distribution/pom.xml [hunk 5]
```diff
@@ -46,142 +46,107 @@
...
+                                    <arguments>
+                                        <argument>-classpath</argument>
+                                        <classpath/>
+                                        <argument>-Ddruid.extensions.loadList=[]</argument>
+                                        <argument>-Ddruid.extensions.directory=${project.build.directory}/extensions
+                                        </argument>
+                                        <argument>
+                                            -Ddruid.extensions.hadoopDependenciesDir=${project.build.directory}/hadoop-dependencies
+                                        </argument>
+                                        <argument>io.druid.cli.Main</argument>
```
Comment (jihoonson) [inline distribution/pom.xml]: We have changed all package names from `io.druid` to `org.apache.druid`. Please fix this. 

Same for other places below.

**Comment by fjy** [comment]
@asdf2014 can we finish this up?

**Pull request merged**